src/NLog/MessageTemplates/ValueFormatter.cs

```diff
@@ -286,16 +315,23 @@ private bool SerializeCollectionObject(IEnumerable collection, string format, IF
 
                 if (separator) builder.Append(", ");
 
-                if (item is string || !(item is IEnumerable))
-                    FormatObject(item, format, formatProvider, builder);
-                else
-                    SerializeWithoutCyclicLoop((IEnumerable)item, format, formatProvider, builder, objectsInPath, depth + 1);
+                SerializeCollectionItem(item, format, formatProvider, builder, ref objectsInPath, depth);
 
                 separator = true;
             }
             return true;
         }
 
+        private void SerializeCollectionItem(object item, string format, IFormatProvider formatProvider, StringBuilder builder, ref SingleItemOptimizedHashSet<object> objectsInPath, int depth)
+        {
+            if (item is IConvertible convertible)
+                SerializeConvertibleObject(convertible, format, formatProvider, builder);
+            else if (!(item is IEnumerable))
```

Comment:
please change to `if (item is IEnumerable enumerable))` (and switch else cases)


thanks

Reply:
Done